python/lsst/meas/base/compensatedGaussian/_compensatedTophat.py

```diff
@@ -0,0 +1,221 @@
+        return cls.FLUX_ORDER
+
+    def __init__(
+        self,
+        config: SingleFrameCompensatedTophatFluxConfig,
+        name: str,
+        schema,
+        metadata,
+        logName=None,
+        **kwds,
+    ):
+        super().__init__(config, name, schema, metadata, logName, **kwds)
+
+        flagDefs = FlagDefinitionList()
+
+        self.aperture_keys = {}
+        self._rads = {}
+        self._inner_scale = config.inner_scale
+        self._outer_scale = config.outer_scale
+        for aperture in config.apertures:
+            base_key = f"{name}_{aperture}"
+
+            # flux
+            flux_str = f"{base_key}_instFlux"
+            flux_key = schema.addField(
+                flux_str,
+                type="D",
+                doc="Compensated Tophat flux measurement.",
+                units="count",
+            )
+
+            # flux error
+            err_str = f"{base_key}_instFluxErr"
+            err_key = schema.addField(
+                err_str,
+                type="D",
+                doc="Compensated Tophat flux error.",
+                units="count",
+            )
+
+            # mask bits
+            mask_str = f"{base_key}_mask_bits"
+            mask_key = schema.addField(mask_str, type=np.int32, doc="Mask bits set within aperture.")
+
+            # failure flags
+            failure_flag = flagDefs.add(f"{aperture}_flag", "Compensated Tophat measurement failed")
+            oob_flag = flagDefs.add(f"{aperture}_flag_bounds", "Compensated Tophat out-of-bounds")
+
+            self.aperture_keys[aperture] = (flux_key, err_key, mask_key, failure_flag, oob_flag)
+            self._rads[aperture] = int(math.ceil(self._outer_scale*aperture))
+
+        self.flagHandler = FlagHandler.addFields(schema, name, flagDefs)
+        self._max_rad = max(self._rads)
+
+    def fail(self, measRecord, error=None):
+        """Record failure
+
+        See also
+        --------
+        lsst.meas.base.SingleFramePlugin.fail
+        """
+        if error is None:
+            self.flagHandler.handleFailure(measRecord)
+        else:
+            self.flagHandler.handleFailure(measRecord, error.cpp)
+
+    def measure(self, measRecord, exposure):
+        center = measRecord.getCentroid()
+        bbox = exposure.getBBox()
+
+        y = center.getY() - bbox.beginY
+        x = center.getX() - bbox.beginX
+
+        y_floor = math.floor(y)
+        x_floor = math.floor(x)
+
+        ctrl = ApertureFluxAlgorithm.Control()
+
+        for aperture, (flux_key, err_key, mask_key, failure_flag, oob_flag) in self.aperture_keys.items():
+            rad = self._rads[aperture]
+
+            if Point2I(center) not in exposure.getBBox().erodedBy(rad):
+                self.flagHandler.setValue(measRecord, failure_flag.number, True)
+                self.flagHandler.setValue(measRecord, oob_flag.number, True)
+                continue
+
+            y_slice = slice(y_floor - rad, y_floor + rad + 1, 1)
```

Comment:
Since we checked the bounding box we know this will work.  Same code is used in the CompensatedGaussians, and these failure modes are also tested.  I'll add a comment.